pkg/controller/dnszone/zonereconciler.go

```diff
@@ -495,6 +624,51 @@ func (zr *ZoneReconciler) addRateLimitingStatusEntries() {
 	zr.dnsZone.Status.LastSyncTimestamp = &tmpTime
 }
 
+func parentLinkRecordName(dnsZoneName string) string {
+	return fmt.Sprintf("%s-ns", dnsZoneName)
+}
+
+func lookupSOARecord(zone string, logger log.FieldLogger) (bool, error) {
+	// TODO: determine if there's a better way to obtain resolver endpoints
+	clientConfig, _ := dns.ClientConfigFromFile("/etc/resolv.conf")
+	client := dns.Client{Timeout: dnsClientTimeout}
+
+	dnsServers := []string{}
+	for _, s := range clientConfig.Servers {
+		dnsServers = append(dnsServers, fmt.Sprintf("%s:%s", s, clientConfig.Port))
+	}
+	logger.WithField("servers", dnsServers).Debug("looking up domain SOA record")
+
+	m := &dns.Msg{}
+	m.SetQuestion(zone+".", dns.TypeSOA)
+	for _, s := range dnsServers {
+		in, rtt, err := client.Exchange(m, s)
+		if err != nil {
+			logger.WithError(err).WithField("server", s).Debug("query for SOA record failed")
+			continue
+		}
+		log.WithField("server", s).Debugf("SOA query duration: %v", rtt)
+		if len(in.Answer) > 0 {
+			for _, rr := range in.Answer {
+				soa, ok := rr.(*dns.SOA)
+				if !ok {
+					logger.Debugf("Record returned is not an SOA record: %#v", rr)
+					continue
+				}
+				if soa.Hdr.Name != (zone + ".") {
```

Comment:
Consider checking if zone already has a trailing period and don't add one if it does.